plaso/storage/redis/writer.py

```diff
@@ -0,0 +1,357 @@
+    """Prepares a task storage for merging.
+
+    Args:
+      task (Task): task.
+    """
+   # No preparations are necessary.
+
+  def AddAnalysisReport(self, analysis_report):
+    """Adds an analysis report.
+
+    Args:
+      analysis_report (AnalysisReport): a report.
+    """
+    self._store.AddAnalysisReport(analysis_report)
+
+  def AddEvent(self, event, serialized_data=None):
+    """Adds an event.
+
+    Args:
+      event(EventObject): an event.
+      serialized_data (Optional[bytes]): serialized form of the event.
+    """
+    self._store.AddEvent(event, serialized_data=serialized_data)
+
+  def AddEventData(self, event_data, serialized_data=None):
+    """Adds an event data.
+
+    Args:
+      event_data(EventData): an event.
+      serialized_data (Optional[bytes]): serialized form of the event data.
+    """
+    self._store.AddEventData(event_data, serialized_data=serialized_data)
+
+  def AddEventSource(self, event_source, serialized_data=None):
+    """Adds an event source.
+
+    Args:
+      event_source (EventSource): an event source.
+      serialized_data (Optional[bytes]): serialized form of the event source.
+    """
+    self._store.AddEventSource(event_source, serialized_data=serialized_data)
+
+  def AddEventTag(self, event_tag, serialized_data=None):
+    """Adds an event tag.
+
+    Args:
+      event_tag (EventTag): an event tag.
+      serialized_data (Optional[bytes]): serialized form of the event tag.
+    """
+    self._store.AddEventTag(event_tag, serialized_data=serialized_data)
+
+  def AddWarning(self, warning, serialized_data=None):
+    """Adds a warning.
+
+    Args:
+      warning (ExtractionWarning): a warning.
+      serialized_data (Optional[bytes]): serialized form of the warning.
+    """
+    self._store.AddWarning(warning, serialized_data=serialized_data)
+
+  def CheckTaskReadyForMerge(self, task):
+    """Checks if a task is ready for merging into the store.
+
+    Args:
+      task (Task): task.
+
+    Returns:
+      bool: True if the task is ready to be merged.
+    """
+    return self._store.IsFinalized()
+
+  def GetEventDataByIdentifier(self, identifier):
+    """Retrieves specific event data.
+
+    Args:
+      identifier (AttributeContainerIdentifier): event data identifier.
+
+    Returns:
+      EventData: event data or None if not available.
+    """
+    return self._store.GetEventDataByIdentifier(identifier)
+
+  def GetEvents(self):
+    """Retrieves the events.
+
+    Returns:
+      generator(EventObject): event generator.
+    """
+    return self._store.GetEvents()
+
+  # pylint: disable=redundant-returns-doc,useless-return
+  def GetFirstWrittenEventSource(self):
+    """Retrieves the first event source that was written after open.
+
+    Using GetFirstWrittenEventSource and GetNextWrittenEventSource newly
+    added event sources can be retrieved in order of addition.
+
+    Returns:
+      EventSource: event source or None if there are no newly written ones.
```

Comment:
Maybe change to:
`EventSource: None since there are no newly written event sources.`